### ISSUE TITLE
Error when sending email through swoosh

### ISSUE THREAD

**Issue by gabrielrinaldi** [gen-smtp/gen_smtp#205]
When trying to use [swoosh](https://github.com/swoosh/swoosh) and this library on master I get a conflicted version for `ranch`. 

If I force version `1.7.1` for `ranch` and build from `gen_smtp` master I get the following error:
```
[error] Task #PID<0.2251.0> started from #PID<0.2250.0> terminating
** (BadMapError) expected a map, got: []
    (stdlib) maps.erl:187: :maps.get(:content_type_params, [], [])
    (gen_smtp) /Users/gabriel/Code/Aero/Backend/deps/gen_smtp/src/mimemail.erl:664: :mimemail.ensure_content_headers/7
    (gen_smtp) /Users/gabriel/Code/Aero/Backend/deps/gen_smtp/src/mimemail.erl:158: :mimemail.encode/2
    (swoosh) lib/swoosh/adapters/smtp.ex:45: Swoosh.Adapters.SMTP.deliver/2
    (aero) lib/aero_web/api/resolvers/user.ex:50: anonymous fn/8 in AeroWeb.API.Resolvers.User.create_user/3
    (elixir) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
    (elixir) lib/task/supervised.ex:35: Task.Supervised.reply/5
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Function: #Function<0.33802017/0 in AeroWeb.API.Resolvers.User.create_user/3>
    Args: []
```

The error does not happen when using the published `0.15` version. 

I am happy to provide more information if you need.

**Comment by mworrell** [comment]
Looks like there is a list passed where the new (1.x) library expects a map.

**Comment by gabrielrinaldi** [comment]
Do you have a release date for 1.0 yet? I can try to fix it once we are close, will probably have to open a PR on the swoosh side in our case.

**Comment by mworrell** [comment]
The 1.0 should be really-soon-now.

I think this is just a case of the format of some options, need to check the changes made regarding the move from lists to maps.

**Comment by seriyps** [comment]
might be related to #190

**Comment by seriyps** [comment]
So indeed, the format of `mimemail:mimetuple()` have changed in #190.
It used to be
```
{
                 mime_type(),
                 mime_subtype(),
                 headers(),
                 Parameters :: list(),
                 Body :: <..>
                }
```
and now it is
```
{
                 mime_type(),
                 mime_subtype(),
                 headers(),
                 Parameters :: map(),
                 Body :: <..>
                }
```
So, I guess swoosh should be updated.
Another option is to make gen_smtp accept lists and convert them to maps transparently.

**Comment by arjan** [comment]
The only place this is used in Swoosh is [here](https://github.com/swoosh/swoosh/blob/db1b0aff4a4bf43fa235c28bc7e657c56a20939d/lib/swoosh/adapters/smtp/helpers.ex#L18). I'll make a PR to address this on their side, it's a minor change.

**Comment by arjan** [comment]
So let's release 1.0 and then I'll create a swoosh pr.

**Comment by Shemeikka** [comment]
Hi,

If I understood this correctly, then this might be fixed in [Swoosh v1.1.2](https://github.com/swoosh/swoosh/pull/553).
